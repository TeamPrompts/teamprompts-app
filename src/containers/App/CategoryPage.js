import React from 'react';
import { useParams } from 'react-router-dom';
import Filters from '../../components/Filters/Filters';
import FillInTheBlanks from '../../FillInTheBlanks/FillInTheBlanks';
import { pageTypes } from '../../constants';
import Breadcrumbs from './Breadcrumbs';
import Heading from './Heading';
import tagBySlug from './tagBySlug';

function fitbsByTag({ fitbs, tag, tags }) {
  return fitbs.filter(fitb => fitb.tags.includes(tag.id));
}

function CategoryPage({ fitbs, history, tags }) {
  const { slug } = useParams();
  const tag = tagBySlug({ slug, tags });
  return (
    <>
      <Breadcrumbs pageType={pageTypes.CategoryPage} tag={tag} />
      <Heading title="TeamPrompts" />
      <Filters
        allOff={true}
        filter={{ name: tag.name }}
        fitbs={fitbs}
        onClick={console.log}
        tags={tags}
      />
      <ul>
        {fitbsByTag({ fitbs, tag, tags }).map(fitb => (
          <li key={fitb.id}>
            <FillInTheBlanks fitb={fitb} history={history} tag={tag} />
          </li>
        ))}
      </ul>
    </>
  );
}

export default CategoryPage;
