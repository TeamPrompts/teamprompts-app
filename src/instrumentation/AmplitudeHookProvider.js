import { Amplitude, AmplitudeProvider } from '@amplitude/react-amplitude';
import amplitude from 'amplitude-js';
import React, { createContext, useContext } from 'react';

const { REACT_APP_USE_AMPLITUDE } = process.env;

export const AmplitudeContext = createContext();

export function useAmplitude() {
  return useContext(AmplitudeContext);
}

const mockProps = { logEvent: console.log };

function AmplitudeHookProvider({ apiKey, children }) {
  return (
    <AmplitudeProvider
      amplitudeInstance={amplitude.getInstance()}
      apiKey={apiKey}
    >
      <Amplitude>
        {props => {
          return (
            <AmplitudeContext.Provider
              value={REACT_APP_USE_AMPLITUDE === 'true' ? props : mockProps}
            >
              {children}
            </AmplitudeContext.Provider>
          );
        }}
      </Amplitude>
    </AmplitudeProvider>
  );
}

export default AmplitudeHookProvider;
