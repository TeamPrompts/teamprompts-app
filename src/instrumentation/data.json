{
  "device_id": "a5a25d38-d067-4e9d-8ed6-622de26c1a4aR",
  "user_id": null,
  "timestamp": 1576972626961,
  "event_id": 278,
  "session_id": 1576972532372,
  "event_type": "view collection",
  "version_name": null,
  "platform": "Web",
  "os_name": "Chrome",
  "os_version": "79",
  "device_model": "Mac",
  "device_manufacturer": null,
  "language": "en-US",
  "carrier": null,
  "api_properties": {},
  "event_properties": {
    "name": "teamprompts-app",
    "nodeEnv": "production",
    "version": "0.19.0",
    "collectionName": "all",
    "loadTime": "TBD",
    "promptsDisplayed": "TBD",
    "promptsInCollection": 41
  },
  "user_properties": {},
  "uuid": "a0827d06-1dee-48a2-8810-75c4b3724977",
  "library": {
    "name": "amplitude-js",
    "version": "5.7.0"
  },
  "sequence_number": 278,
  "groups": {},
  "group_properties": {},
  "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36"
}
