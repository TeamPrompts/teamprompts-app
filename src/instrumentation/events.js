import { name, version } from '../../package.json';
import getType from './getType';

const { NODE_ENV } = process.env;

const base = {
  appName: name,
  appVersion: version,
  loadTime: 'TBD',
  nodeEnv: NODE_ENV
};

export const copyPrompt = {
  properties: function properties({ content, fitb, viewPosition }) {
    return {
      ...base,
      content,
      interactionContext: 'TBD', // Q: can we get this from the previous event?
      promptId: fitb.id,
      promptType: getType(fitb),
      viewPosition
    };
  },
  type: 'copy prompt'
};

export const filterByCollection = {
  properties: function properties({ tag }) {
    return {
      ...base,
      collectionId: tag.id,
      collectionName: tag.name,
      interactionContext: 'TBD' // Q: can we get this from the previous event?
    };
  },
  type: 'filter by collection'
};

export const navigateToPromptDetail = {
  properties: function properties({ fitb, viewPosition }) {
    return {
      ...base,
      promptId: fitb.id,
      viewPosition
    };
  },
  type: 'navigate to prompt detail'
};

export const viewCollection = {
  properties: function properties({ counter, tag, url }) {
    return {
      ...base,
      collectionId: tag.id,
      collectionName: tag.name,
      promptsDisplayed: 'TBD',
      promptsInCollection: counter,
      url
    };
  },
  type: 'view collection'
};

export const viewHome = {
  properties: function properties({ url }) {
    return {
      ...base,
      url
    };
  },
  type: 'view home'
};

export const viewPromptDetail = {
  properties: function properties({ fitb, tag, url }) {
    return {
      ...base,
      collectionId: tag.id,
      collectionName: tag.name,
      promptId: fitb.id,
      promptType: getType(fitb),
      url
    };
  },
  type: 'view prompt detail'
};
