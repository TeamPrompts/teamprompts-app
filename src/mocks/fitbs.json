[
  {
    "author": "John Cutler",
    "examples": [
      "limiting work in progress",
      "it meant that at times people would have nothing to do",
      "limiting WIP made it more possible to work on what was important"
    ],
    "id": "recevUpbRFDbVHx7L",
    "prompts": ["aspect of how you work", "reason", "realization"],
    "source": "I used to find __BLANK__ to be highly counterintuitive. It seemed destined to fail because __BLANK__. I now see things differently. The big shift was realizing that __BLANK__.",
    "tags": ["recTqD26ElVvXKVad"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "increasing",
      "speed of onboarding",
      "increases",
      "rate of overall adoption",
      "their ability to see value earlier"
    ],
    "id": "recglWG0JkW6S4z5T",
    "prompts": [
      "increasing/decreasing",
      "some activity",
      "increases/decreases",
      "key performance metric, financial metric, etc.",
      "factors, caveats, assumptions"
    ],
    "source": "We believe that __BLANK__ __BLANK__ causes __BLANK__ in some __BLANK__. There are a number of assumptions/factors underpinning that causal relationship including __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "id": "recD5UagMbU0oW464",
    "prompts": [
      "some opportunity",
      "some intervention",
      "know/be more confident",
      "some leading indicator",
      "some approach to de-risking"
    ],
    "source": "If you look at our near-time three month roadmap, you will see a number of bets. The riskiest bet, with the highest potential upside, is our bet to focus on __BLANK__ by trying to __BLANK__. We will __BLANK__ this was the right bet if we observe __BLANK__. We can actively de-risk this bet by __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "id": "recCWoOnmN4aEi5dT",
    "prompts": [
      "some sea-change in technology adoption and/or some shift in consumer behavior",
      "fast/slower",
      "forcing/allowing",
      "some activity",
      "forcing/allowing",
      "some other activity"
    ],
    "source": "The writing is on the wall about __BLANK__. What is unclear is timing. As a company, we are betting that this will occur __BLANK__ than expected, which will influence our strategy by __BLANK__ us to __BLANK__, and __BLANK__ us to __BLANK__.",
    "tags": ["recmdwXtLaj7hls73", "rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "to add steps to the workflow",
      "more customers completing the existing workflow"
    ],
    "id": "recQ2jud83D1dCxr4",
    "source": "To feel more confident about our decision to __BLANK__, I would like to observe __BLANK__.",
    "tags": ["recG0ANQWrp3SRvvY"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "spend too long deliberating over the behavior of the page filter",
      " test early with customers"
    ],
    "id": "recrYoglrdF32klOv",
    "source": "I’m concerned that if we __BLANK__, we may miss the opportunity to __BLANK__.",
    "tags": ["rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "the commodification of storage",
      "the rate of dropping costs",
      "1) COGS would not drop as fast as we expect"
    ],
    "id": "recm6xCbOJSJZ6jwY",
    "prompts": ["an assumption", "that assumption", "a list of ways"],
    "source": "We seem to be treating __BLANK__ as a foregone conclusion. I’m not sure it is that simple. If we were proven wrong (or less right) about __BLANK__ , it would impact us in the following ways: __BLANK__  .",
    "tags": ["recmdwXtLaj7hls73"]
  },
  {
    "author": "John Doe",
    "examples": ["this amazing challenge"],
    "id": "recqeN7TDq4kdhFah",
    "source": "To even know where to start, we would need to better understand __BLANK__.",
    "tags": ["recG0ANQWrp3SRvvY"]
  },
  {
    "author": "The Person",
    "examples": [
      "this huge problem we have",
      "feel the wrath of all that has come before"
    ],
    "id": "rec99C9lxMl6aSedq",
    "source": "A danger of not resolving __BLANK__ now is that we will eventually __BLANK__.",
    "tags": ["recTqD26ElVvXKVad"]
  },
  {
    "author": "Mary Tine",
    "examples": [
      "how we perceive user onboarding",
      "our projections might be off"
    ],
    "id": "reckkQSV8NuHKOAHr",
    "prompts": ["some assumption", "reason"],
    "source": "The weakest link in our mental model is __BLANK__. If we’re off -- even by a little -- it could have a big impact on this initiative because __BLANK__.",
    "tags": ["rechbGc1FLmwxb9Hg"]
  },
  {
    "author": "John Cutler",
    "examples": [
      "understand the EMEA customer persona",
      "we haven't really made EMA a focus",
      "we decide to shift into EMEA"
    ],
    "id": "receM4Gus7HH43oLM",
    "prompts": ["an action", "reason", "new event"],
    "source": "We may be over-confident in our ability to __BLANK__. It hasn’t really been a problem yet because __BLANK__, but it could become a problem in the future if __BLANK__.",
    "tags": ["recmdwXtLaj7hls73"]
  }
]
