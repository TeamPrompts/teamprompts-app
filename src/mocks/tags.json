[
  {
    "fitbs": ["recNsahDzKCx0IQPk"],
    "id": "recmdxpSo7MSWEg00",
    "name": "1:1s",
    "slug": "1-1s"
  },
  {
    "fitbs": ["recQ2jud83D1dCxr4", "recqeN7TDq4kdhFah"],
    "id": "recG0ANQWrp3SRvvY",
    "name": "decision making",
    "slug": "decision-making"
  },
  { "id": "recEZjB6WTnRt6aon", "name": "interview", "slug": "interview" },
  {
    "fitbs": [
      "reckkQSV8NuHKOAHr",
      "recBKiB7u7DLP6QPe",
      "recrYoglrdF32klOv",
      "recCWoOnmN4aEi5dT",
      "recD5UagMbU0oW464",
      "recglWG0JkW6S4z5T"
    ],
    "id": "rechbGc1FLmwxb9Hg",
    "name": "measurement",
    "slug": "measurement"
  },
  {
    "fitbs": ["recnGYhq8E9LeRPlW", "recNsahDzKCx0IQPk"],
    "id": "recBxMIOsQqBhGSaQ",
    "name": "product management",
    "slug": "product-management"
  },
  {
    "fitbs": [
      "rec99C9lxMl6aSedq",
      "rec590aP1VRIE8f23",
      "recNsahDzKCx0IQPk",
      "recevUpbRFDbVHx7L"
    ],
    "id": "recTqD26ElVvXKVad",
    "name": "retrospectives",
    "slug": "retrospectives"
  },
  {
    "id": "rec1f3sLCcKszXSi4",
    "name": "reviewing decisions",
    "slug": "reviewing-decisions"
  },
  {
    "fitbs": [
      "receM4Gus7HH43oLM",
      "recBKiB7u7DLP6QPe",
      "recm6xCbOJSJZ6jwY",
      "recCWoOnmN4aEi5dT",
      "recD5UagMbU0oW464",
      "recglWG0JkW6S4z5T"
    ],
    "id": "recmdwXtLaj7hls73",
    "name": "strategy",
    "slug": "strategy"
  }
]
