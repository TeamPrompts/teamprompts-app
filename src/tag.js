const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
