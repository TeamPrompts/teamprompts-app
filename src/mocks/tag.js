import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
