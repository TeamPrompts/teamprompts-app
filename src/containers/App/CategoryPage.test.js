import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { useAmplitude } from '../../instrumentation/AmplitudeHookProvider';
import fitbs from '../../mocks/fitbs';
import tag from '../../mocks/tag';
import tags from '../../mocks/tags';
import CategoryPage from './CategoryPage';

const mockSlug = tag.slug;

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    slug: mockSlug
  })
}));

describe('CategoryPage', () => {
  it('to match snapshot', () => {
    const tree = create(
      <Router>
        <CategoryPage fitbs={fitbs} match={{ url: `/tag.slug` }} tags={tags} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('tagEmpty', () => {
    const { logEvent } = useAmplitude();

    render(
      <Router>
        <CategoryPage fitbs={fitbs} match={{ url: `/tag.slug` }} tags={[]} />
      </Router>
    );

    expect(logEvent).not.toBeCalled();
  });
});
