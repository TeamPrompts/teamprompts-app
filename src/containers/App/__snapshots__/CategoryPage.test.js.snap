// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage to match snapshot 1`] = `
Array [
  <div
    className="mt-2"
  >
    <a
      className="text-indigo-700 underline"
      href="#/"
      onClick={[Function]}
    >
      Home
    </a>
     &gt; 
    <span
      className="capitalize"
    >
      retrospectives
    </span>
  </div>,
  <h1
    className="font-bold mb-8 mt-2 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>,
  <ul
    className="flex flex-wrap mb-4"
  >
    <li
      className="mx-1 my-1"
    >
      <a
        href="#/retrospectives"
        onClick={[Function]}
      >
        <button
          className="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
          data-testid="retrospectives"
          onClick={[Function]}
          type="button"
        >
          retrospectives (1)
        </button>
      </a>
    </li>
  </ul>,
  <ul>
    <li>
      <div
        className="border-b border-gray-500 leading-loose py-8 text-lg"
        onClick={[Function]}
      >
        Do you know the way to 
        <span
          className="inline-flex flex-col leading-none"
        >
          <div
            className="self-center"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              className=""
              data-testid="input"
              onChange={[Function]}
              placeholder="________________"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "1px",
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              ________________
            </div>
          </div>
          <label
            className="self-center text-sm"
          >
            place
          </label>
        </span>
        ?
        <span
          className="inline-block ml-3 relative"
        >
          <input
            checked={false}
            className="hidden"
            data-testid="receM4Gus7HH43oLM"
            id="receM4Gus7HH43oLM"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="cursor-pointer"
            htmlFor="receM4Gus7HH43oLM"
          >
            <div
              className="w-6 h-3 bg-gray-400 rounded-full shadow-inner"
            />
            <div
              className="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0"
            />
          </label>
        </span>
        <button
          className="bg-blue-500 focus:outline-none font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-sm text-white"
          data-testid="copy-button"
          onClick={[Function]}
          type="button"
        >
          Ctrl+C
        </button>
      </div>
    </li>
  </ul>,
]
`;
